service/src/main/java/uk/gov/hmcts/reform/fpl/jobs/UpcomingHearingsFinder.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.fpl.jobs;
+
+import lombok.extern.slf4j.Slf4j;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationEventPublisher;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.UpcomingHearingsFound;
+import uk.gov.hmcts.reform.fpl.service.calendar.CalendarService;
+import uk.gov.hmcts.reform.fpl.service.search.SearchService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.time.ZoneOffset.UTC;
+
+@Slf4j
+public class UpcomingHearingsFinder implements Job {
+
+    private static final String HEARING_DATE_PROPERTY = "data.hearingDetails.value.startDate";
+
+    @Value("${UPCOMING_HEARINGS_DAYS:2}")
+    private int noticeDaysBeforeHearing = 2;
+
+    @Autowired
+    private CalendarService calendarService;
+
+    @Autowired
+    private SearchService searchService;
+
+    @Autowired
+    private ApplicationEventPublisher applicationEventPublisher;
+
+    @Override
+    public void execute(JobExecutionContext jobExecutionContext) {
+        final String jobName = jobExecutionContext.getJobDetail().getKey().getName();
```

Comment:
Wouldn't it be more beneficial to just shortcut the whole elastic search when the tooggle is not enabled?

Reply:
not sure, i would like to have quicker feedback if it works on prod even if toggle is off. In such situation i guess I should change the way emails are send. Now this class emits events which makes in unaware from who and why is consuming event, and is detached from any handler of event. Handler decide what to do with event. With suggested changes i should use direct call of notification handler

Reply:
That's good point 👍